### PR TITLE
Add logdata to some rules

### PR DESCRIPTION
**Pull request by fzipi**

### CODE REVIEW

**Comment by dune73** [comment]
Thank you for the PR @fzipi. Is "some rules" a random selection or will you do the other rules, too?

**Comment by fzipi** [comment]
@dune73 I'm going to do all rules, but there are some rules that are more "boolean", in the sense that if they match, a particular argument equals 1 or 0. Take for example 920280: it will match if `&REQUEST_HEADERS:Host "@eq 0"`.. so there is no much to log there. Makes sense to have logdata in those cases? And if it makes, what are we going to log?

**Comment by dune73** [comment]
I see your point. No, there is nothing really to log in such cases, I think. But I think it is worthwhile to do all the rest. Would that be part of this PR or have you already done all the non-boolean ones here?

**Comment by fgsch** [comment]
If possible, I'd like for this PR to only cover the missing logdata and not the matched_var case (the case change).

**Comment by fzipi** [comment]
@fgsch You are perfectly right. I got those case changes back. Maybe we should add another issue with that, now I feel we missed vars casing in our 'Rule Cleanup'....

@dune73 Anyway, now I think we have all logdata covered. I made all non booleans here, but please, someone let me know if I'm short on adding logdata...

File: rules/REQUEST-949-BLOCKING-EVALUATION.conf [hunk 1]
```diff
@@ -48,6 +48,7 @@ SecRule TX:ANOMALY_SCORE "@ge %{tx.inbound_anomaly_score_threshold}" \
     t:none,\
     log,\
     msg:'Inbound Anomaly Score Exceeded (Total Score: %{TX.ANOMALY_SCORE})',\
+    logdata:'%{TX.ANOMALY_SCORE}',\
```
Comment (fgsch) [inline rules/REQUEST-949-BLOCKING-EVALUATION.conf]: I don't think this is needed. The score is in the message already.

File: rules/REQUEST-920-PROTOCOL-ENFORCEMENT.conf [hunk 2]
```diff
@@ -475,6 +477,7 @@ SecRule TX:CRS_VALIDATE_UTF8_ENCODING "@eq 1" \
     block,\
     t:none,\
     msg:'UTF8 Encoding Abuse Attack Attempt',\
+    logdata:'%{matched_var}',\
```
Comment (fgsch) [inline rules/REQUEST-920-PROTOCOL-ENFORCEMENT.conf]: Have we verified this work with validateUtf8Encoding?

File: rules/REQUEST-920-PROTOCOL-ENFORCEMENT.conf [hunk 3]
```diff
@@ -754,6 +759,7 @@ SecRule REQUEST_HEADERS:Content-Length "!@rx ^0$" \
     block,\
     t:none,\
     msg:'Request Containing Content, but Missing Content-Type header',\
+    logdata:'%{matched_var}',\
```
Comment (fgsch) [inline rules/REQUEST-920-PROTOCOL-ENFORCEMENT.conf]: I don't think this is needed.

Comment (fzipi) [reply rules/REQUEST-920-PROTOCOL-ENFORCEMENT.conf]: Erased.

File: rules/REQUEST-949-BLOCKING-EVALUATION.conf (hunk 1 above)
Comment (fzipi) [reply rules/REQUEST-949-BLOCKING-EVALUATION.conf]: Erased.

**Comment by dune73** [comment]
Looks good to me. Did not check for detailed coverage, though.

Booleans like 920380 and 920410 are covered now. Others are not. Maybe we should make this systematic and define it in the contribution guide.

What do you guys think?

**Comment by fgsch** [comment]
@dune73 I agree about adding this to the contribution guide. 

fwiw, planning to take a look at this PR tomorrow.

**Comment by fgsch** [comment]
Looks good to me. 

Can we squash the 4 commits before it's merged?

**Comment by dune73** [comment]
Sorry, I do not get you. Can you explain please?

**Comment by fgsch** [comment]
Sure. There are 4 commits in these PR. It'd be good if they are merged so only 1 commit appears in the history.

**Comment by dune73** [comment]
Ah, I see. You mean the 4 commits in this PR #980 here?

**Comment by fgsch** [comment]
Correct. Keeping the history clean is quite useful IMO.

**Comment by dune73** [comment]
I support you there. However, my n00b git skills block me here. How would I perform that consolidation?

**Comment by fgsch** [comment]
I normally do:

1. `git rebase -i -p <hash from the last commit before the first commit on this PR>`
2. mark all but the first commit as `f(ixup)`
3. save
4. force push

Sometimes you also want to edit the original commit message to reflect some of the changes in the squashed commits. In that case you can also change the first one to `r(eword)` to amend it.

Please note that you can shoot yourself really badly with rebase. 
However, as long as you don't force push the branch and the original code is in a remote, you can do this operation as many times as you want after you reset the branch, in this case by doing `git reset --hard remotes/origin/pr/980`.

**Comment by dune73** [comment]
Thanks.

Ideally, @fzipi would do that before we merge, or do I get this wrong?

**Comment by fgsch** [comment]
Ideally yes. Anyone with write perms can do it, though.

**Comment by dune73** [comment]
@fzipi could you try and do this? I'll keep the bandage ready.

**Comment by fzipi** [comment]
@dune73 @fgsch It's easier than that: you just select 'Squash and merge' from the github interface, correct the commit text, and that's it!

**Comment by dune73** [comment]
Now that's a very anti-climaxic move! I had the popcorn ready and now this!

**Comment by fzipi** [comment]
Sorry. Next time I will take the time for that ;)

For completion, I've made a screenshot for you 🙂 

github-rebase

**Comment by fzipi** [comment]
BTW, on the command line I usually do a `git merge --squash <branch>`. In this case:

```
git checkout v3.1/dev
git merge --squash add-logdata # <-- name of the branch
```
Then it fires your favourite editor for changing the commit text.

**Comment by dune73** [comment]
Thank you. This sounds a lot less exciting than @fgsch's approach, but I admit I don't enjoy shooting my foot very much, so that's probably the path I am going to take in the future. :)

**Comment by fgsch** [comment]
@fzipi that's not the same, and hence I did not mention it.
It will not create a merge commit, just merge them directly but if you are happy with it, wfm.

**Comment by fzipi** [comment]
@fgsch Yeah, I know. For our purposes, it will work. You can use the third option in the image, and do a rebase and merge, as you mentioned it.

Maybe we can discuss this on the next agenda, with the only purpose of using the same solution each time. I think that many times we merge multiple commits than can be rebased/squashed to have a cleaner history.

**Comment by fgsch** [comment]
Going back to the remaining issue from this PR, is the preference to use uppercase for the variables?

**Comment by fzipi** [comment]
I would say yes. But if someone has another opinion....

**Comment by dune73** [comment]
Contributing says the following:

### Variables naming conventions
* Variable names are lowercase using chars from [a-z0-9_]
* To somewhat reflect the fact that the syntax for variable usage is different when you define it (using setvar) and when you use it, we propose the following visual distinction:
  * Lowercase letters for collection, dot as separator, variable name. E.g: setvar:tx.foo_bar_variable
  * Capital letters for collection, colon as separator, variable name. E.g: SecRule TX:foo_bar_variable

**Pull request merged**